diff -ur gnuradio-3.7.0rc0_orig/CMakeLists.txt gnuradio-3.7.0rc0/CMakeLists.txt
--- gnuradio-3.7.0rc0_orig/CMakeLists.txt	2013-06-21 14:36:57.000000000 +0100
+++ gnuradio-3.7.0rc0/CMakeLists.txt	2013-06-26 20:56:25.084149118 +0100
@@ -16,7 +16,8 @@
 # along with GNU Radio; see the file COPYING.  If not, write to
 # the Free Software Foundation, Inc., 51 Franklin Street,
 # Boston, MA 02110-1301, USA.
-
+link_libraries(m)
+link_libraries(pthread)
 ########################################################################
 if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
     message(FATAL_ERROR "Prevented in-tree build. This is bad practice.")
