--- CMakeLists.txt_orig	2012-01-23 23:08:20.187632154 +0000
+++ CMakeLists.txt	2012-01-23 23:07:31.598327261 +0000
@@ -93,9 +93,9 @@
 set(GR_PKG_DATA_DIR     ${GR_DATA_DIR}/${CMAKE_PROJECT_NAME})
 set(GR_DOC_DIR          ${GR_DATA_DIR}/doc)
 set(GR_PKG_DOC_DIR      ${GR_DOC_DIR}/${CMAKE_PROJECT_NAME}-${DOCVER})
-set(GR_CONF_DIR         etc)
+set(GR_CONF_DIR         ../etc)
 set(GR_PKG_CONF_DIR     ${GR_CONF_DIR}/${CMAKE_PROJECT_NAME}/conf.d)
-set(GR_LIBEXEC_DIR      libexec)
+set(GR_LIBEXEC_DIR      lib${LIB_SUFFIX})
 set(GR_PKG_LIBEXEC_DIR  ${GR_LIBEXEC_DIR}/${CMAKE_PROJECT_NAME})
 set(GRC_BLOCKS_DIR      ${GR_PKG_DATA_DIR}/grc/blocks)
 
